src/transformers/pipelines/automatic_speech_recognition.py

```diff
@@ -496,6 +504,15 @@ def _forward(self, model_inputs, return_timestamps=False, generate_kwargs=None):
             else:
                 generate_kwargs["encoder_outputs"] = encoder(inputs, attention_mask=attention_mask)
 
+            if self.processor:
+                # Added initial prompt for whisper
+                if "initial_prompt" in generate_kwargs:
+                    initial_prompt = generate_kwargs.pop("initial_prompt")
+                    generate_kwargs["prompt_ids"] = self.processor.get_prompt_ids(initial_prompt)
```

Comment:
By design, we can get the same behaviour using the `tokenizer` method `get_prompt_ids`:
```python
generate_kwargs["prompt_ids"] = self.tokenizer.get_prompt_ids(initial_prompt)
```
Let's simplify this logic to only ever have the feature extractor + tokenizer, and always rely on `tokenizer.get_prompt_ids` to convert the prompt.